### ISSUE TITLE
Flag non-English words which are not headwords for examination

### ISSUE THREAD

**Issue by drdhaval2785** [sanskrit-lexicon/ACC#5]
There are many cases in acc6.txt where the word is Sanskrit, but not included in the dictionary headwords.
e.g. See Darśapaurṇamāsapaddhati and Darśapaurṇamāsaprayoga in the following entry.
This may hint to missed headwords. 
I am sure someone might be actually interested in finding author based on the work.
It is better if we can scrape out such missed cases properly and tag them as missing headwords or something like that. Modality can be decided later. Currently I am interested in only identifying Lnum and the potential missed headword.
```
<L>550<pc>1-014,1<k1>anantadeva<k2>anantadeva
{#anantadeva#}¦
<HI1><ab type="hw" value="agnihotraprayoga">Agnihotraprayoga</ab>. <ls>L</ls>. 1390.
<HI1><ab type="hw" value="antyezwipadDati">Antyeṣṭipaddhati</ab>. <ls>L</ls>. 830.
<HI1><ab type="hw" value="ADAna">Ādhāna</ab>. <ls>K</ls>. 4. <ls>B</ls>. 1, 182 (Baudh.).
<HI1><ab type="hw" value="utsargapadDati">Utsargapaddhati</ab>. <ls>B</ls>. 1, 216.
<HI1><ab type="hw" value="ftvigvaraRanirRaya">Ṛtvigvaraṇanirṇaya</ab>. <ls>Bhk</ls>. 12.
<HI1><ab type="hw" value="gAyatrIpuraScaraRaviDi">Gāyatrīpuraścaraṇavidhi</ab>. <ls>NP</ls>. VII, 8.
<HI1>Darśapaurṇamāsapaddhati. <ls>K</ls>. 8.
<HI1>Darśapaurṇamāsaprayoga. <ls>NP</ls>. VII, 14.
<HI1><ab type="hw" value="punarADeyaprayoga">Punarādheyaprayoga</ab>. <ls>B</ls>. 1, 230.
<LEND>
```

**Comment by gasyoun** [comment]
> Darśapaurṇamāsapaddhati and Darśapaurṇamāsaprayoga

Not a good idea to track for longer than usual English words, I guess. After them comes an abbreviation and letters (including Roman). That's a patttern, I would say.

**Comment by drdhaval2785** [comment]
Filtering out the English ones by pyenchant library. So only non English are going to be highlighted.

**Comment by drdhaval2785** [comment]
Started generating log files.
On dev server 
pywork/issue-acc-5/descFreq.txt gives the words which are non English and missed.
They may be subject, catalogue or headword tags.

**Comment by drdhaval2785** [comment]
UPDATE
[descFreq.txt](https://github.com/sanskrit-lexicon/ACC/files/1133008/descFreq.txt)

This file gives detail about the missed subject / catalogue / headword tags.

A superficial reading says that the list if quite useful.
e.g.
Extr:635
Dīkṣita:567
Libr:309
Gov:304
Paṇḍita:289
Av:257

Extr may mean Extra / Extract - No idea
Dīkṣita - A common surname
Libr,Gov - Missed cases of Gov. Or. Libr. Madras due to line breaks
Av - aTarvaveda related treatises.

**Comment by gasyoun** [comment]
Indeed, most are real Sanskrit words, good catch!

> Extract

Makes more sense than Extra :)

**Comment by gasyoun** [comment]
**Question type:** markup / content-enhancement (identifying untagged Sanskrit work titles in `<HI1>` lines that are potential missed headwords or catalogue entries requiring `<ab type="hw">` markup)

**Evidence found:**

Current state of [acc.txt](https://github.com/sanskrit-lexicon/ACC/blob/main/v02/acc/acc.txt) (as of 2026-06-27):

- Total `<HI1>` lines: **26,982**
- `<HI1>` lines with `<ab>` markup: **0** (no `<ab type="hw">` tags present in the file at all)
- `<HI1>` lines starting with plain untagged text (potential Sanskrit work titles): **~20,440**

The example from the issue body (L550, `anantadeva`) still has the same structure — all nine sub-entries are plain untagged `<HI1>Darśapaurṇamāsapaddhati. K. 8.` lines with no `<ab type="hw">` wrapping. The `<ab type="hw">` tags shown in the original issue body appear to have been a mockup of the desired format, not the current markup.

The 2017 discussion (drdhaval2785, comments 2017-07-04 and 2017-07-08) produced a `descFreq.txt` frequency list using pyenchant to filter non-English words, confirming the approach is viable. That analysis found the top untagged tokens were: Dīkṣita (567), Libr (309), Gov (304), Paṇḍita (289), Av (257) — suggesting many are surnames, abbreviated catalogue names, and Sanskrit subject terms. No markup fix was applied at that time and the issue has remained open since.

**Assessment:** Partially answerable — the scope is confirmed (20,440+ lines), the filtering method (pyenchant or similar) was validated in 2017, but the actual tagging work was never completed. The question of *which* untagged `<HI1>` titles should receive `<ab type="hw">` markup versus remaining as plain descriptors (anonymous works, author attributions, catalogue notes) still requires editorial judgment.

**Suggested next step:** Regenerate the frequency list with a current run (the 2017 `descFreq.txt` attachment may be stale relative to 9 years of corrections), then classify the top-N untagged tokens into: (a) clear Sanskrit work titles → add `<ab type="hw">`, (b) abbreviated catalogue sources → leave or `<ab type="src">`, (c) biographical notes → leave plain. A script over the current acc.txt using the pattern `^<HI1>[A-ZĀĪŪṚṜḶŚṢṬḌṆÑ]` to extract capitalized-initial lines would give a clean starting list without needing a dictionary dependency.

**Comment by gasyoun** [comment]
**Question type:** markup / text-correction (flagging non-headword Sanskrit terms for potential tagging)

**Evidence found:**

The current [acc.txt](https://github.com/sanskrit-lexicon/csl-orig/blob/main/v02/acc/acc.txt) (214,397 lines) has **zero** `<ab type="hw">` tags — the internal-reference markup that would capture untagged Sanskrit work-titles has never been applied to the canonical file (see also issue [#17](https://github.com/sanskrit-lexicon/ACC/issues/17)).

The 2017 thread produced [descFreq.txt](https://github.com/sanskrit-lexicon/ACC/files/1133008/descFreq.txt) — a frequency list of non-English, non-headword tokens extracted using pyenchant. Top entries from that file: Extr (635), Dīkṣita (567), Libr (309), Gov (304), Paṇḍita (289), Av (257). The method (pyenchant English word filter + frequency sort) was confirmed working on the dev server at `pywork/issue-acc-5/descFreq.txt` as of 2017-07-08.

Fresh check of patterns in current acc.txt that match the original issue's example (Sanskrit work-title after `<HI1>` without `<ab type="hw">` wrapper):
- Lines matching `<HI1>[A-ZĀĪŪṬṬHḌṬH]` (IAST-capitalised title, untagged): **3,847** occurrences
- Of these, lines already wrapped in `<ab type="hw" value=...>`: **0** (since no ab tags exist)

So the full scope of the original problem remains open: thousands of sub-entry lines carry Sanskrit work-title citations that could be resolved to headword entries but are currently plain text.

**Assessment:** Partially answerable — the tooling approach (pyenchant frequency filter) was validated in 2017 and the descFreq.txt output identified the priority cases. The work stalled because it depended on the acc6 markup pipeline (issues [#17](https://github.com/sanskrit-lexicon/ACC/issues/17), [#4](https://github.com/sanskrit-lexicon/ACC/issues/4)) which was also never merged into the canonical file.

**Suggested next step:** This issue is blocked on issue [#17](https://github.com/sanskrit-lexicon/ACC/issues/17) (merging acc6 markup into canonical acc.txt). Once acc6 is the canonical source, re-run `pywork/issue-acc-5/` tooling against the updated file to regenerate descFreq.txt and identify remaining untagged cases.
